### PR TITLE
use 2-legged client auth

### PR DESCRIPTION
**Pull request by klauern**
As per the documentation, 2-legged OAuth2 is preferred if you're not running as a third party.  As this is a provided feature, it's simpler to set up and requires one less parameter.

### CODE REVIEW

**Comment by hbagdi** [comment]
@klauern, API provides certain endpoints which have to be accessed with a authenticated user (eg- like a photo, add a photo) so it is necessary to use the staticsource method for automated testing.

I'm not sure if I'm making sense.

**Comment by klauern** [comment]
That's fine, but you authenticate with the `clientcredentials.Config`, right?  At least, the way I'm testing it, I'm getting mostly working with this, and my application configuration has this set.

image

Is this not workable?  I'm no OAuth expert, so you're quite possibly right.  That said, I had some issues using the `StaticToken....` that you had, and I found this work instead, at least from a client-testing perspective.

**Comment by hbagdi** [comment]
clientcredentials.Config can be used and it'll most probably work with endpoints which don't need user authentication but will fail otherwise.

Do the following-
```bash
$ cat "unsplash_usertoken=<YOUR-AUTH-TOKEN>" > auth.env
```
If you don't have an auth token you can generate one with some work. I can help you with that. Let me know.t

The method you propose is the way to go but here we have constraints for test automation.

File: unsplash/unsplash_test.go [hunk 1]
```diff
@@ -61,6 +66,14 @@ func getAppAuth() *AuthConfig {
 	return &config
 }
 
+func randCollectionName(n int) string {
```
Comment (hbagdi) [inline unsplash/unsplash_test.go]: How about just 'randName`? Generalizing will help in future.

**Comment by klauern** [comment]
Hello.  I teased out the commit to another branch, and then created another pull request.  Let me know if that works for you.

File: unsplash/unsplash_test.go (hunk 1 above)
Comment (klauern) [inline unsplash/unsplash_test.go]: easy enough.  This is referenced in #26

**Comment by hbagdi** [comment]
Closing this for now, reopen if we need further discussion.

**Pull request closed without merging**